pipeline-runner/R/qc-7-embed_and_cluster.R

```diff
@@ -5,189 +5,19 @@
 
 embed_and_cluster <- function(scdata, config, sample_id, task_name = 'configureEmbedding') {
 
-  scdata.embedding <- run_embedding(scdata, config)
-  scdata.embedding <- run_clustering(scdata.embedding, config)
-  scdata.embedding <- coloring_samples_and_cluster(scdata.embedding)
-
-  cells_order <- rownames(scdata.embedding@meta.data)
-
-  # Create a generic pattern with the UMAP coordinates
-  embeddingPreview <- unname(purrr::map2(
-    scdata.embedding@reductions$umap@cell.embeddings[cells_order, 1],
-    scdata.embedding@reductions$umap@cell.embeddings[cells_order, 2],
-    function(x, y) {
-      c("x" = x, "y" = y)
-    }
-  ))
-
-  # Create plot for cellsets
-  embeddingPreviewByCellSets <- purrr::map2(
-    embeddingPreview,
-    unname(scdata.embedding@active.ident[cells_order]),
-    function(x, y) {
-      append(x, c("cluster" = paste("Cluster", y, sep = " ")))
-    }
-  )
-
-  # Adding color for cellsets
-  embeddingPreviewByCellSets <- purrr::map2(
-    embeddingPreviewByCellSets,
-    unname(scdata.embedding@meta.data[cells_order, "color_active_ident"]),
-    function(x, y) {
-      append(x, c("col" = y))
-    }
-  )
-
-  # Create plot for samples
-  embeddingPreviewBySamples <- purrr::map2(
-    embeddingPreview,
-    unname(scdata.embedding@meta.data[cells_order, "samples"]),
-    function(x, y) {
-      append(x, c("sample" = y))
-    }
-  )
-
-  # Adding color for samples
-  embeddingPreviewBySamples <- purrr::map2(
-    embeddingPreviewBySamples,
-    unname(scdata.embedding@meta.data[cells_order, "color_samples"]),
-    function(x, y) {
-      append(x, c("col" = y))
-    }
-  )
-
-  # Create plot for MT-content
-  embeddingPreviewMitochondrialContent <- purrr::map2(
-    embeddingPreview,
-    unname(scdata.embedding@meta.data[cells_order, "percent.mt"]),
-    function(x, y) {
-      append(x, c("mt-content" = y))
-    }
-  )
-  # Create plot for Doublet-score
-  embeddingPreviewDoubletScore <- purrr::map2(
-    embeddingPreview,
-    unname(scdata.embedding@meta.data[cells_order, "doublet_scores"]),
-    function(x, y) {
-      append(x, c("doublet-score" = y))
-    }
-  )
-
   plots <- list()
-  plots[generate_gui_uuid("", task_name, 0)] <- list(embeddingPreviewByCellSets)
-  plots[generate_gui_uuid("", task_name, 1)] <- list(embeddingPreviewBySamples)
-  plots[generate_gui_uuid("", task_name, 2)] <- list(embeddingPreviewMitochondrialContent)
-  plots[generate_gui_uuid("", task_name, 3)] <- list(embeddingPreviewDoubletScore)
+  plots[generate_gui_uuid("", task_name, 0)] <- list()
+  plots[generate_gui_uuid("", task_name, 1)] <- list()
+  plots[generate_gui_uuid("", task_name, 2)] <- list()
+  plots[generate_gui_uuid("", task_name, 3)] <- list()
```

Comment:
Are these necessary? Does anything happen if we ommit?

Reply:
I've tried it and it broke stuff, so I preferred letting them survive. This will be resolved first in api/UI and then I'll come back here to delete these. Still need to create the API ticket

Reply:
I don't think it broke stuff. Maybe something else was broken that you encountered. Look at this:

```R
guidata <- list()
guidata['a'] <- list()
guidata['b'] <- list()
all.equal(list(), guidata)
# [1] TRUE
guidata
# list()
```

Reply:
I see what you mean